posixlib/src/main/resources/scala-native/time.c

```diff
@@ -1,3 +1,5 @@
+#define __USE_BSD /* for tm_gmtoff and tm_zone */
```

Comment:
#define __USE_XOPEN // POSIX XSI - tm_gmtoff, tm_zone, strptime()

Reply:
@LeeTibbert let me add and this one, but CentOS 7 for example requires `__USE_BSD ` or `_GNU_SOURCES`

Reply:
I think that `__USE_XOPEN` is closer to the concept that we want to
express.  Operating systems/C_compilers which do not have that should
ignore it and use 36_byte/9_field definition.

Reply:
@LeeTibbert for example `CentOS Linux release 7.9.2009 (Core)` has:
```
/* Used by other time functions.  */
struct tm
{
  int tm_sec;                   /* Seconds.     [0-60] (1 leap second) */
  int tm_min;                   /* Minutes.     [0-59] */
  int tm_hour;                  /* Hours.       [0-23] */
  int tm_mday;                  /* Day.         [1-31] */
  int tm_mon;                   /* Month.       [0-11] */
  int tm_year;                  /* Year - 1900.  */
  int tm_wday;                  /* Day of week. [0-6] */
  int tm_yday;                  /* Days in year.[0-365] */
  int tm_isdst;                 /* DST.         [-1/0/1]*/

# ifdef __USE_BSD
  long int tm_gmtoff;           /* Seconds east of UTC.  */
  const char *tm_zone;          /* Timezone abbreviation.  */
# else
  long int __tm_gmtoff;         /* Seconds east of UTC.  */
  const char *__tm_zone;        /* Timezone abbreviation.  */
# endif
};
```

Without `__USE_BSD` it fails :(

Reply:
Ah, the joys of cross Operating System support.  Arrgh!   

I seem to remember that issues such as these drove me to memset in PR #1667.

What do you think of something like
```
// POSIX XSI - tm_gmtoff, tm_zone, strptime()
#define __USE_XOPEN // POSIX XSI - many Linux
#define __USE_BSD // CentOS Linux release 7.9.2009 (Core)
// end XSI
```
We will probably have to define for other operating systems as we encounter them. What
does MacOS use, any idea?

Reply:
@LeeTibbert nothing.

For macOS and OpenBSD no define is required.

Reply:
@LeeTibbert where is `__USE_XOPEN` required for it?